twisted/conch/ssh/transport.py

```diff
@@ -1498,6 +1593,68 @@ def ssh_KEXINIT(self, packet):
                     self._dhMaximalGroupSize,
                     ))
 
+    def _ssh_KEX_ECDH_REPLY(self, packet):
+        """
+        Called to handle a reply to a ECDH exchange message(KEX_ECDH_INIT).
+
+        Like the handler for I{KEXDH_INIT}, this message type has an
+        overlapping value.  This method is called from C{ssh_KEX_DH_GEX_GROUP}
+        if that method detects a non-group key exchange is in progress.
+
+        Payload::
+
+            string serverHostKey
+            string server Elliptic Curve Diffie-Hellman public key
+            string signature
+
+        We verify the host key and continue if it passes verificiation.
+        Otherwise raise an exception and return.
+
+        @type packet: L{bytes}
+        @param packet: The message data.
+
+        @return: A deferred firing when key exchange is complete.
+        """
+
+        # Get the host public key, the raw ECDH public key bytes and the signature
+        hostKey, f, signature, packet = getNS(packet, 3)
+
+        fingerprint = b':'.join([binascii.hexlify(ch) for ch in
+                                 iterbytes(md5(hostKey).digest())])
+        d = self.verifyHostKey(hostKey, fingerprint)
+        d.addCallback(self._continue_KEX_ECDH_REPLY, hostKey, f, signature)
+        d.addErrback(
+            lambda unused: self.sendDisconnect(
+                DISCONNECT_HOST_KEY_NOT_VERIFIABLE, b'bad host key'))
+        return d
+
+    def _continue_KEX_ECDH_REPLY(self, ignored, hostKey, pktPub, signature):
+        #Save off the host public key.
+        self.theirECHost = hostKey
```

Comment:
I could have sworn I made this comment before, but:

1. `theirECHost` is a public attribute based on its name; it doesn't need to be exposed to application programmers.
2. It actually doesn't need to be recorded at all in an attribute; it's used only in the local function.
3. Same for `theirECPub`.

Can you get rid of these attributes unless there's a reason to keep them?  And if there's a reason to keep them, add some test coverage that checks for them?